Fix des outliers@420 shinyvalidate@main

gogonzo opened the pull request:
- turn `categorical_var`  to use generic data-extract-spec instead of handling filter_spec only. Removed custom handling of the NA in the module. Theoretically, this NA problem could exist in every module where user could provide similar settings.
- No validation on the filter - we can't expect specific filter selection in the data_extract_spec - consequently we could do the same in the rest of the modules.

chlebowa commented:
Input validation is broken now:
![image](https://user-images.githubusercontent.com/114988527/211273555-faceb3c9-fcdd-4874-8c2c-11307211c81e.png)

chlebowa commented:
Also, I get `object 'summary_table' not found` in the console but I cannot seem to reproduce it, not sure where it comes from.

gogonzo commented:
@chlebowa I think validation isn't broken but has been turned off for the filter. The idea of having des is that one can specify any variable with select_spec and apply any filter using filter_spec. The way how it was specified in tm_distribution was against this logic. Categorical variables were selected using filter_spec instead of select_spec which lead to BUG.
Please see alternative (valid) example which omits validation on the `420_shinyvalidate@main`
```r
pkgload::load_all("teal.modules.general")
library(scda)
ADSL <- synthetic_cdisc_data("latest")$adsl

fact_vars_adsl <- names(Filter(isTRUE, sapply(ADSL, is.factor)))
app <- init(
  data = cdisc_data(
    cdisc_dataset("ADSL", ADSL, code = "ADSL <- synthetic_cdisc_data(\"latest\")$adsl"),
    check = TRUE
  ),
  modules = modules(
    tm_outliers(
      outlier_var = list(
        data_extract_spec(
          dataname = "ADSL",
          select = select_spec(
            label = "Select variable:",
            choices = variable_choices(ADSL, c("AGE", "BMRKR1")),
            selected = "AGE",
            multiple = FALSE,
            fixed = FALSE
          )
        )
      ),
      categorical_var = data_extract_spec(
        dataname = "ADSL",
        select = select_spec(
          choices = variable_choices(ADSL, fact_vars_adsl),
          selected = variable_choices(ADSL, fact_vars_adsl)[1]
        )

      ),
      ggplot2_args = teal.widgets::ggplot2_args(
        labs = list(subtitle = "Plot generated by Outliers Module")
      )
    )
  )
)

shinyApp(app$ui, app$server)
```

See now other valid example which has filter and select for the categorical var. Please select different variable for a select
```r
pkgload::load_all("teal.modules.general")
library(scda)
ADSL <- synthetic_cdisc_data("latest")$adsl

fact_vars_adsl <- names(Filter(isTRUE, sapply(ADSL, is.factor)))
vars <- choices_selected(variable_choices(ADSL, fact_vars_adsl))
app <- init(
  data = cdisc_data(
    cdisc_dataset("ADSL", ADSL, code = "ADSL <- synthetic_cdisc_data(\"latest\")$adsl"),
    check = TRUE
  ),
  modules = modules(
    tm_outliers(
      outlier_var = list(
        data_extract_spec(
          dataname = "ADSL",
          select = select_spec(
            label = "Select variable:",
            choices = variable_choices(ADSL, c("AGE", "BMRKR1")),
            selected = "AGE",
            multiple = FALSE,
            fixed = FALSE
          )
        )
      ),
      categorical_var = data_extract_spec(
        dataname = "ADSL",
        filter = filter_spec(
          vars = vars,
          choices = value_choices(ADSL, vars$selected),
          selected = value_choices(ADSL, vars$selected),
          multiple = TRUE
        ),
        select = select_spec(
          choices = variable_choices(ADSL, fact_vars_adsl),
          selected = variable_choices(ADSL, fact_vars_adsl)[1]
        )

      ),
      ggplot2_args = teal.widgets::ggplot2_args(
        labs = list(subtitle = "Plot generated by Outliers Module")
      )
    )
  )
)

shinyApp(app$ui, app$server)
```

With that I suggest to not validate filter_spec which is optional and keep using select_spec to select categorical_var

mhallal1 reviewed (commented):
Testing apps with `cat_var` with `data_extract_spec` having the 3 cases below:
1) only `filter_spec`
2) only `select_spec`
3) both of the above

The logic makes sense especially when having only a `filter_spec` (case 1) and we get the result below:
![Screenshot from 2023-01-09 17-08-08](https://user-images.githubusercontent.com/86970066/211355295-4867319a-71a4-47d8-8627-e0bf844df3bc.png)

mhallal1 reviewed (commented):
please remove the obsolete definition below:
`is_cat_filter_spec <- inherits(categorical_var[[1]]$filter[[1]], "filter_spec")`
